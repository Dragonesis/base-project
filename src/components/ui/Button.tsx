import React, { FC, ReactNode } from 'react'
import styled from '@emotion/styled'

export interface ButtonProps {
  handleClick?: () => void
  children: ReactNode
  type?: 'button' | 'submit'
}

export const Button: FC<ButtonProps> = ({ type = 'button', children, handleClick }) => {
  return (
    <Core
      type={type}
      onClick={handleClick ? handleClick : undefined}
    >
      {children}
    </Core>
  )
}

const Core = styled.button`
  height: 24px;
  border-radius: 4px;
  border: 1px solid #3a3a3a;
  padding-left: 10px;
  padding-right: 10px;
`
