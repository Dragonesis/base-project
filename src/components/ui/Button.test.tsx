import React from 'react'
import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

import { Button } from './Button'

it('Button renders correctly', () => {
  const { asFragment } = render(<Button>Button</Button>)
  expect(asFragment()).toMatchSnapshot()
})

it('Button children test', () => {
  render(<Button type='submit'>Button</Button>)
  expect(screen.getByText('Button')).toBeInTheDocument()
})

it('Button type test', () => {
  render(
    <>
      <Button type='button'>Button</Button>
      <Button type='submit'>Submit</Button>
    </>
  )

  expect(screen.getByText('Button')).toHaveAttribute('type', 'button')
  expect(screen.getByText('Submit')).toHaveAttribute('type', 'submit')
})

it('Button callback test', () => {
  const handleClick = jest.fn()
  render(<Button type='submit' handleClick={handleClick}>Button</Button>)
  const button = screen.getByText('Button')
  expect(button).toBeInTheDocument()
  userEvent.click(button)
  expect(handleClick).toHaveBeenCalled()
})
