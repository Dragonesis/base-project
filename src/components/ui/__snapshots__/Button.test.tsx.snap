// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  height: 24px;
  border-radius: 4px;
  border: 1px solid #3a3a3a;
  padding-left: 10px;
  padding-right: 10px;
}

<button
    class="emotion-0"
    type="button"
  >
    Button
  </button>
</DocumentFragment>
`;
